contracts/utils/structs/EnumerableSet.sol

```diff
@@ -442,6 +488,24 @@ library EnumerableSet {
         }
     }
 
+    /**
+     * @dev Removes all the values from a set. O(n).
+     *
+     * WARNING: Developers should keep in mind that this function has an unbounded cost and using it may render the
+     * function uncallable if the set grows to the point where clearing it consumes too much gas to fit in a block.
+     */
+    function clear(Bytes32x2Set storage self) internal {
+        bytes32[2][] storage v = self._values;
+
+        uint256 len = length(self);
+        for (uint256 i = 0; i < len; ++i) {
+            delete self._positions[_hash(v[i])];
+        }
+        assembly ("memory-safe") {
+            sstore(v.slot, 0)
+        }
```

Comment:
We don't have a version of `Arrays.unsafeSetLength` for `bytes32[2][]`.

Do we want to add that (quite specific) version to Arrays.sol, of are we ok with this line of assembly here?

Reply:
I'm fine with the assembly here.